app/javascript/src/components/Dashboard/Notes/NewNoteForm.jsx

```diff
@@ -15,30 +17,54 @@ export default function NewNoteForm({ onClose, refetch }) {
       logger.error(err);
     }
   };
+
   return (
     <Formik
       initialValues={{
         title: "",
+        tag: "internal",
         description: "",
+        assignedContact: contacts[0].value,
+        dueDate: new Date(),
+        doesNoteHaveDueDate: false,
       }}
       onSubmit={handleSubmit}
       validationSchema={yup.object({
         title: yup.string().required("Title is required"),
+        tag: yup.string().required("Tag is required"),
         description: yup.string().required("Description is required"),
+        assignedContact: yup.string().required("Assignee is a required field"),
+        doesNoteHaveDueDate: yup.boolean(),
+        dueDate: yup.string().when("doesNoteHaveDueDate", {
+          is: true,
+          then: yup.string().required("Due date is required"),
+        }),
       })}
     >
-      {({ isSubmitting }) => (
+      {({ isSubmitting, values }) => (
         <Form>
           <Input label="Title" name="title" className="mb-6" />
-          <Textarea label="Description" name="description" rows={8} />
+          <Select options={issueTypes} name="tag" className="mb-6" />
+          <Textarea
+            label="Description"
+            name="description"
+            rows={8}
+            className="mb-6"
+          />
+          <Select options={contacts} name="assignedContact" className="mb-6" />
+          <Switch
+            name="doesNoteHaveDueDate"
+            label="Add Due Date to Note"
+            className="mb-6"
+          />
+          {values?.doesNoteHaveDueDate && <DateInput name="dueDate" />}
```

Comment:
Rather than using `mb-6` for each of the blocks individually, please use `space-y-6`.

Please refer [space between](https://tailwindcss.com/docs/space#class-reference) from tailwindCSS for more details.